### PR TITLE
Update instructions to install dependencies using pip

### LINKED ISSUE #222: Command "mamba" is not found
At least three users reported that the `mamba info` step in the instructions just brought up an error that the mamba command is not found.

The solution is to specify the full path to mamba, e.g. `$HOME/bin/mamba info`, or `$HOME/bin/mamba init` to initialize the user's shell.

### PR DESCRIPTION
**Pull request by stargaser**
Switch to relying on installing dependencies with pip instead of conda.

* Use a minimal `environment.yml` to set the environment name and Python version, and install astroscrappy
* Update package versions in `requirements.txt`
* Update the version checks in `check_env.py`
* Update the Dockerfile used for Binder
* Update the CircleCI setup
* Update the instructions to install Miniconda, activate the environment and then do `pip install -r requirements.txt`
* Update the emails with workshop date and known facilitators
* Fix some imports in the main Cosmology notebook for astropy 6.0

Fixes #222

### CODE REVIEW

File: 00-Install_and_Setup/requirements.txt [hunk 1]
```diff
@@ -20,11 +19,11 @@ jupyterlab
...
 nbgitpuller
-astropy>=5.0
+astropy>=5.3
 asdf
 asdf-astropy
 gwcs>=0.16
 photutils>=1.3.0
 specutils>=1.1.1
-astroquery
+astroquery --pre
```
Comment (bmorris3) [inline 00-Install_and_Setup/requirements.txt:28]: Is there a known issue that we're avoiding with `--pre`?

Comment (stargaser) [reply 00-Install_and_Setup/requirements.txt:28]: @bsipocz recommends to always install astroquery this way. They are on a continuous release model. I think the idea is that services are changing all the time, so they make releases quite infrequently and one should always get the pre-release version.

**Comment by stargaser** [comment]
Darn it, I ran into a problem installing astroscrappy with pip. I think it needs to be installed with conda.

**Comment by stargaser** [comment]
@bmorris3 I've got everything updated and running on binder.org. I don't understand the CI failure.

**Comment by stargaser** [comment]
Just saw #226 but in this PR I am reverting back to miniconda...

**Comment by mwcraig** [comment]
I think `miniconda` is either using mamba behind the scenes or will be soon. In any event, these steps should ensure that `conda` uses `libmamba` for the solve:

```
conda install -n base conda-libmamba-solver
conda config --set solver libmamba
```

This worked for the AAVSO folks who already had Anaconda installed. Source doc is [here](https://www.anaconda.com/blog/a-faster-conda-for-a-growing-community).

**Comment by mwcraig** [comment]
(totally fine with #226 just getting closed....)

**Comment by stargaser** [comment]
conda is using the libmamba-solver starting from [v23.10.0 onwards](https://conda.org/blog/2023-11-06-conda-23-10-0-release/), so I just put in the instructions to check the version and do `conda update conda`. Looks like this change came a few days after the AAVSO workshop.

**Comment by mwcraig** [comment]
Excellent! One of the AAVSO folks was trying to use conda to solve the environment and it hadn't solved after an hour so we did the libconda thing. Was really painful...

**Comment by stargaser** [comment]
@bmorris3 I found a fix to the strange CI problem that started happening in the second half of October. I've squashed all my commits too. Would you please review?

**Review by bmorris3 — APPROVED** [review]
Thanks @stargaser!

**Pull request merged**